File: productservice/src/main/java/com/productservice/command/data/Product.java
package com.productservice.command.data;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Table(name = "product")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Product {
    @Id
    private String id;

    @Column(name = "name", nullable = false, length = 250)
    private String name;

    @Column(name = "price")
    private Float price;

    @Column(name = "thumbnail", length = 500)
    private String thumbnail;

    @Column(name = "description")
    private String description;

    @Column(name = "stock")
    private int stock;

    @Column(name = "user_id")
    private String userId;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

}

File: frontEnd/src/app/model/product.ts
import { ProductImage } from "./productImage";

export interface Product {
    id : number

    name : string;

    price : number;

    thumbnail : string;

    description : string;

    stock: number;

    categoryId : number ;

    userId: number;

    url : string;

    product_images: ProductImage[];
}